1.10/test/run

```diff
@@ -174,44 +188,83 @@ test_connection() {
   return 1
 }
 
+test_connection_s2i() {
+  cat $cid_file
+  info "Testing the HTTP connection (http://$(container_ip):${test_port})"
+
+  if test_for_output "http://$(container_ip):${test_port}/" "NGINX is working" &&
+     test_for_output "http://$(container_ip):${test_port}/" "NGINX2 is working" localhost2 &&
+     test_for_output "http://$(container_ip):${test_port}/nginx-cfg/default.conf" "404"; then
+     return 0
+  fi
+
+  return 1
+}
+
 test_application() {
   # Verify that the HTTP connection can be established to test application container
-  run_test_application &
+  run_test_application "test-app" &
 
   # Wait for the container to write it's CID file
   wait_for_cid
 
   test_scl_usage "nginx -v" "nginx version: nginx/1.10."
-  check_result $?
+  check_result $? "test-app"
 
   test_connection
-  check_result $?
+  check_result $? "test-app"
   cleanup_test_app
 }
 
+test_pre_init_script() {
```

Comment:
Definitely +1 for passing app name as argument to functions. So be able to use these functions in more cases.

`test_pre_init_script` and `test_application`
`test_connection_s2i` and `test_connection`
are very similar. Only functions used for s2i app testing adds 1-2 lines of code.

I suggest:
    1. remove `cleanup_test_app` from `test_application` and have to call it explicitly after testing of each app finished.
    (2. rename `test_application` to `test_application_basics`)
    3. when testing s2i app use same `test_application` and before calling `cleanup_test_app` do

```
test_command "cat /opt/app-root/etc/nginx.d/default.conf | grep 'resolver' | sed -e 's/resolver\(.*\);/\1/' | grep 'DNS_SERVER'" ""
check_result $? "pre-init-test-app"

test_for_output "http://$(container_ip):${test_port}/aliased/index2.html" "NGINX2 is working" 
```

Reply:
+1, will implement this. Just a note, it's the other way around, `test_connection` has one more line to test. But i got your point, and I'll fix my code.

Reply:
I haven't changed this in #41, because I'd rather see using the `test.lib.sh` from common repo.